.github/workflows/test_with_clone.yaml

```diff
@@ -0,0 +1,291 @@
+
+
+            - name: Give DE ownership of all objects in synapse, synapse_raw, schemachange
+              shell: bash
+              run: |
+                # TODO: Why REVOKE current grants? This might cause trouble.
+                snow sql -q "USE ROLE SECURITYADMIN; GRANT OWNERSHIP ON DATABASE ROLE ${SNOWFLAKE_DOMAIN_CLONE_SANITIZED}.synapse_all_admin TO ROLE ${SNOWFLAKE_CLONE_PROXY_ADMIN_ROLE} REVOKE CURRENT GRANTS;"
+                snow sql -q "USE ROLE SECURITYADMIN; GRANT DATABASE ROLE ${SNOWFLAKE_DOMAIN_CLONE_SANITIZED}.synapse_all_admin TO ROLE ${SNOWFLAKE_CLONE_PROXY_ADMIN_ROLE};"
+
+                snow sql -q "USE ROLE SECURITYADMIN; GRANT OWNERSHIP ON DATABASE ROLE ${SNOWFLAKE_DOMAIN_CLONE_SANITIZED}.synapse_raw_all_admin TO ROLE ${SNOWFLAKE_CLONE_PROXY_ADMIN_ROLE} REVOKE CURRENT GRANTS;"
+                snow sql -q "USE ROLE SECURITYADMIN; GRANT DATABASE ROLE ${SNOWFLAKE_DOMAIN_CLONE_SANITIZED}.synapse_raw_all_admin TO ROLE ${SNOWFLAKE_CLONE_PROXY_ADMIN_ROLE};"
+
+                snow sql -q "USE ROLE SECURITYADMIN; GRANT OWNERSHIP ON DATABASE ROLE ${SNOWFLAKE_DOMAIN_CLONE_SANITIZED}.schemachange_all_admin TO ROLE ${SNOWFLAKE_CLONE_PROXY_ADMIN_ROLE} REVOKE CURRENT GRANTS;"
+                snow sql -q "USE ROLE SECURITYADMIN; GRANT DATABASE ROLE ${SNOWFLAKE_DOMAIN_CLONE_SANITIZED}.schemachange_all_admin TO ROLE ${SNOWFLAKE_CLONE_PROXY_ADMIN_ROLE};"
+
+
+            - name: Grant USAGE on schemas to PROXY_ADMIN
+              shell: bash
+              run: |
+                snow sql -q "USE ROLE SECURITYADMIN; GRANT USAGE ON SCHEMA ${SNOWFLAKE_DOMAIN_CLONE_SANITIZED}.SYNAPSE TO ROLE ${SNOWFLAKE_CLONE_PROXY_ADMIN_ROLE};"
+                snow sql -q "USE ROLE SECURITYADMIN; GRANT USAGE ON SCHEMA ${SNOWFLAKE_DOMAIN_CLONE_SANITIZED}.SYNAPSE_RAW TO ROLE ${SNOWFLAKE_CLONE_PROXY_ADMIN_ROLE};"
+                snow sql -q "USE ROLE SECURITYADMIN; GRANT USAGE ON SCHEMA ${SNOWFLAKE_DOMAIN_CLONE_SANITIZED}.SCHEMACHANGE TO ROLE ${SNOWFLAKE_CLONE_PROXY_ADMIN_ROLE};"
+
+
+            - name: Grant USAGE on PROXY_ADMIN to DE
+              shell: bash
+              run: |
+                snow sql -q "USE ROLE SECURITYADMIN; GRANT ROLE ${SNOWFLAKE_CLONE_PROXY_ADMIN_ROLE} TO ROLE ${SNOWFLAKE_CLONE_ROLE};"
+
+
+            ########################################################################################################
+            ########################################################################################################
+            ##################################### RUNNING SCHEMACHANGE #############################################
+            ########################################################################################################
+            ########################################################################################################
+            - name: Run schemachange on the clone as DATA_ENGINEER
```

Comment:
s/DATA_ENGINEER/the clone admin

Since we're using a variable we may as well keep things generic.